examples/pytorch/gcn/gcn_mp.py

```diff
@@ -125,109 +119,32 @@ def main(args):
         data = PubmedGraphDataset()
     else:
         raise ValueError('Unknown dataset: {}'.format(args.dataset))
-
     g = data[0]
-    if args.gpu < 0:
-        cuda = False
-    else:
-        cuda = True
-        g = g.to(args.gpu)
-
+    device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+    g = g.int().to(device)
     features = g.ndata['feat']
     labels = g.ndata['label']
-    train_mask = g.ndata['train_mask']
-    val_mask = g.ndata['val_mask']
-    test_mask = g.ndata['test_mask']
-    in_feats = features.shape[1]
-    n_classes = data.num_labels
-    n_edges = g.number_of_edges()
-    print("""----Data statistics------'
-      #Edges %d
-      #Classes %d
-      #Train samples %d
-      #Val samples %d
-      #Test samples %d""" %
-          (n_edges, n_classes,
-              train_mask.int().sum().item(),
-              val_mask.int().sum().item(),
-              test_mask.int().sum().item()))
-
-    # add self loop
-    g = dgl.remove_self_loop(g)
-    g = dgl.add_self_loop(g)
-    n_edges = g.number_of_edges()
-
+    masks = g.ndata['train_mask'], g.ndata['val_mask'], g.ndata['test_mask']
+    if args.self_loop:
```

Comment:
Let's drop `self_loop` cmdline argument. It is true in most cases. Use DGL's data transform module when creating the dataset:
```python
data = CoraGraphDataset(transform=dgl.transform.AddSelfLoop())
```

Reply:
Sure.